### PR TITLE
fix: adds a short sha representation to modules list command pointing to the repo specific commit

### LINKED ISSUE #1763: Use short git sha in modules output
### Description of feature

Some of the modules commands print the entire git sha to the terminal, which is pretty long. It would be nicer to use just a six character sha instead to make better use of the available screen space.

Whilst we're there, we could also use more rich hyperlinks (`[link=...]`) to point directly to the sources.

### PR DESCRIPTION
**Pull request by swampie**
Fixes #1763

Reduce the sha representation on `modules list` command by taking the first 7 chars and adds the link to the specific commit.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Comment by swampie** [comment]
For the reviewers: I added two links in the table 
one for the repo_url and the other for the specific commit sha.

I also checked other commands but I only found the `list` one to expose the sha to the user. @ewels

**Comment by ewels** [comment]
@nf-core-bot fix linting

**Comment by ewels** [comment]
Nice one, thanks @swampie!

I was thinking that the git sha was in the output of `nf-core modules lint`, but you're right - I can't see it there.

Would be nice to add hyperlinks to `Module name` (to the nf-core modules page) and `File path` (to open the local file in a code editor), but thats' beyond the scope of my old issue 🙄

File: nf_core/modules/modules_repo.py [hunk 1]
```diff
@@ -69,6 +69,12 @@ def __init__(self, remote_url=None, branch=None, no_pull=False, hide_progress=Fa
 
         self.avail_module_names = None
 
+    def gitless_repo(self):
+        gitless_repo_url = self.remote_url
+        if self.remote_url and ".git" in self.remote_url:
+            gitless_repo_url = gitless_repo_url[:-3]
```
Comment (ewels) [inline nf_core/modules/modules_repo.py:75]: `.git`, not just `git`
```suggestion
            gitless_repo_url = gitless_repo_url[:-4]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -79,6 +79,7 @@
 - Patch: handle file not found when it is an added file to a module ([#2771](https://github.com/nf-core/tools/pull/2771))
 - Handle symlinks when migrating pytest ([#2770](https://github.com/nf-core/tools/pull/2770))
 - Add `--profile` parameter to nf-test command ([#2767](https://github.com/nf-core/tools/pull/2767))
+- Reduce the size of the sha representation in the `list` command and links to the specific commit ([#2870](https://github.com/nf-core/tools/pull/2870))
```
Comment (ewels) [inline CHANGELOG.md:82]:
```suggestion
- Reduce the sha length in the `nf-core modules list local` and add links to the specific commit ([#2870](https://github.com/nf-core/tools/pull/2870))
- Add links the nf-core module page and to open the local file in VSCode for module lint results ([#2870](https://github.com/nf-core/tools/pull/2870))
```

**Comment by ewels** [comment]
@nf-core-bot fix linting

**Pull request merged**